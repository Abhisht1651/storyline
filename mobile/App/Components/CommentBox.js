// @flow

import React from 'react'
import { View, Text, Image, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'

// Styles
import styles from './Styles/CommentBoxStyle'
import images from '../Themes/Images'
import DropDown from './Dropdowns/DropDown3'

type Props = {
  name: string,
  commentText: string,
  replyCount: number,
  isReply: boolean,
  boostCount: number,
  selectionHandler: (arg: number) => void,
  onClickBoost: PropTypes.func,
  onClickReply: PropTypes.func,
  time: string
}

const commentBox = (props: Props) => {
  const {
    name,
    commentText,
    replyCount,
    isReply,
    boostCount,
    selectionHandler,
    onClickBoost,
    onClickReply,
    time
  } = props
  const { replyIcon, boostIcon } = images
  return (
    <View style={styles.container}>
      <View style={styles.commentBoxContainer}>
        <View style={styles.commentBoxHeaderContainer}>
          <Text style={styles.nameText}>{name}</Text>
          <View style={styles.headerRight}>
            <Text style={styles.daysText}>{time}</Text>
            <View style={styles.dropDown}>
              <DropDown
                getOption={selectionHandler}
                headerTitle="Options"
                content={['Edit Comment', 'Delete Comment']}
              />
            </View>
          </View>
        </View>
        <View style={styles.commentTextContainer}>
          <Text style={styles.commentText}>{commentText}</Text>
        </View>
        <View style={styles.actionContainer}>
          {!isReply ? (
            <TouchableOpacity onPress={onClickReply} style={styles.replyContainer}>
              <Image source={replyIcon} style={styles.replyIcon} />
              <Text style={styles.actionText}> {replyCount !== 0 ? replyCount : null} Reply</Text>
            </TouchableOpacity>
          ) : null}
          <TouchableOpacity onPress={onClickBoost} style={styles.boostContainer}>
            <Image source={boostIcon} />
            <Text style={styles.actionText}> {boostCount} Boosts</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  )
}

commentBox.defaultProps = {
  name: 'Jammie Rice',
  commentText:
    'When Robert Mueller broke his silence in May, his main point was that his long-awaited report spoke for itself.',
  replyCount: 1,
  isReply: false,
  boostCount: 450,
  selectionHandler: () => {},
  onClickBoost: () => {},
  onClickReply: () => {},
  time: '32 Days'
}

export default commentBox
