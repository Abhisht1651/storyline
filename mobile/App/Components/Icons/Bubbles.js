import React from 'react'
import { View, TouchableOpacity } from 'react-native'

// Styles
import styles from '../Styles/IconStyles/BubblesStyles'
import RavenSVG from '../../Images/Icons/Path-1.svg'
import ShareSVG from '../../Images/Icons/Path-3.svg'
import ReactSVG from '../../Images/Icons/Path.svg'

const Bubbles = () => {
  return (
    <View style={styles.cardBox}>
      <TouchableOpacity>
        <View style={styles.profileIconBox}>
          <ShareSVG />
        </View>
      </TouchableOpacity>

      <TouchableOpacity style={styles.marginBox}>
        <View style={styles.profileIconBox}>
          <RavenSVG />
        </View>
      </TouchableOpacity>

      <TouchableOpacity style={styles.marginBox}>
        <View style={styles.profileIconBox}>
          <ReactSVG />
        </View>
      </TouchableOpacity>
    </View>
  )
}

export default Bubbles
