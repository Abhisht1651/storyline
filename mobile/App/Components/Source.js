// @flow
import React, { useState, useEffect, useRef } from 'react'
import { View, Text, Image, ImageProps } from 'react-native'

// Styles
import styles from './Styles/SourceStyle'
import images from '../Themes/Images'
import {withApollo} from 'react-apollo';
import {getStorylineDetails,getEventSources,getnews,getMultimediaDetails} from '../Containers/StoryLine/my_queries'



// type Props = {
//   title: string,
//   date: string,
//   description: string,
//   storyImage: ImageProps
// }

const storyLineShortestCard = (props) => {
  var [storyLineShortestCard,setFeed]=useState([]);


  const client=props.client;
  if (!client){
    console.log("Client not working")
  }
  var [url,seturl]=useState([]);
  const displaymulti = async (client) =>{
    try{
      const multi_id="9287fdfa-9f1b-11ea-bb37-0242ac130002";
      const {data:datamulti,loading:loadingmulti}= await client.query({query:getMultimediaDetails,variables:{multimedia_id:multi_id}});
      const {url}=datamulti.GetMultimedia;
      seturl(url);
      }
    catch(e){
      console.log(e)
      }
    }
  useEffect(() => {displaymulti(client);}, []);
  const event="1c7a4e9c-bf37-4827-95f0-73878814b571";
  const displaysource = async (client,event) =>{
    try{
      const {data:dataevent, loading:loadingevent}= await client.query({query:getEventSources,variables:{event:event}});
      const {article_id}=dataevent.GetEventArticle[0];
      const {data:datanews, loading:loadingnews}= await client.query({query:getnews,variables:{article_id:article_id}});
      let item={};
      const {publisher,headline,t_update}=datanews.GetNewsArticle[0];
      item['date']=t_update;
      item['title']=publisher
      item['description']=headline
      setFeed(item);
      }
    catch(e){
      console.log(e)
      }
    }
  useEffect(() => {displaysource(client,event);}, []);
  
  const { date,title,description} = storyLineShortestCard
  const { iconSource } = images
  const storyImage= url;
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.sourceText}>{title}</Text>
        <Image source={iconSource} style={styles.iconSource} />
      </View>
      <View style={styles.flexContainer}>
        <View style={styles.imageContainer}>
          <Image source={{uri:storyImage.toString()}} style={styles.image} />
        </View>
        <View style={styles.detailsContainer}>
          <View style={styles.descriptionContainer}>
            <Text style={styles.descriptionText}>{description}</Text>
          </View>
          <View style={styles.timeContainer}>
            <Text style={styles.timeText}>{date}</Text>
          </View>
        </View>
      </View>
    </View>
  )
}

// storyLineShortestCard.defaultProps = {
//   title: 'AAJ TAK',
//   description: 'AS Boris Delivers A Brexit Deal, EU Have Agreedâ€¦',
//   storyImage: images.storyImage,
//   // date: '23 hours ago'
// }

export default withApollo(storyLineShortestCard)