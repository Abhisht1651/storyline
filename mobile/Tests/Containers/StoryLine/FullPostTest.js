// @flow
import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import TestFile from '../../../App/Containers/StoryLine/FullPost'
import Bubbles from '../../../App/Components/Icons/Bubbles'
import DropDown from '../../../App/Components/Dropdowns/DropDown2'
import Source from '../../../App/Components/Source'
import CommentBar from '../../../App/Components/Commentbars/Commentbar2'
import CommentBox from '../../../App/Components/CommentBox'

describe('FullPost Screen', () => {
  it('should render the component itself correctly', () => {
    const wrapper = shallow(<TestFile />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<TestFile />)
    expect(wrapper.find('Text')).toHaveLength(10)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<TestFile />)
    expect(wrapper.find('Image')).toHaveLength(8)
  })
  it('should render the Bubbles Component', () => {
    const wrapper = shallow(<TestFile />)
    expect(wrapper.find(Bubbles)).toHaveLength(1)
  })
  it('should render the DropDown Component', () => {
    const wrapper = shallow(<TestFile />)
    expect(wrapper.find(DropDown)).toHaveLength(1)
  })
  it('should render the Source Component', () => {
    const wrapper = shallow(<TestFile />)
    expect(wrapper.find(Source)).toHaveLength(3)
  })
  it('should render the CommentBar Component', () => {
    const wrapper = shallow(<TestFile />)
    expect(wrapper.find(CommentBar)).toHaveLength(1)
  })
  it('should render the CommentBox Component', () => {
    const wrapper = shallow(<TestFile />)
    expect(wrapper.find(CommentBox)).toHaveLength(5)
  })
})
