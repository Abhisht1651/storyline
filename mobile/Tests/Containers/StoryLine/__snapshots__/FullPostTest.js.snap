// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullPost Screen should render the component itself correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 67.5,
        "width": 750,
      }
    }
  />
  <ScrollViewMock>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 61,
            "justifyContent": "space-between",
            "width": 650,
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
        >
          <Image
            source={
              Object {
                "default": "",
              }
            }
            style={
              Object {
                "height": 13.5,
              }
            }
          />
        </ForwardRef>
        <Bubbles />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#E6ECED",
            "borderBottomWidth": 1,
            "width": 750,
          }
        }
      />
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 435,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 57.5,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "width": 450,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#8F8FAF",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 16,
                "letterSpacing": 0,
                "lineHeight": 27,
                "marginBottom": 5,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            UK exit from the European Union
          </Text>
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 22,
                "letterSpacing": 0,
                "lineHeight": 29,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Brexit: UK exit from the European Union
          </Text>
          <Text
            style={
              Object {
                "color": "#22226B",
                "fontFamily": "BananaGrotesk-Regular",
                "fontSize": 14,
                "letterSpacing": -0.08,
                "lineHeight": 16,
                "marginTop": 3,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Sept 21, 2019, 5:00PM ET
          </Text>
          <Text
            style={
              Object {
                "color": "#8F8FAF",
                "fontFamily": "BananaGrotesk-Regular",
                "fontSize": 12,
                "letterSpacing": -0.07,
                "marginTop": 5,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Updated 
            2m
             ago
          </Text>
        </View>
        <View
          style={
            Object {
              "marginBottom": 80,
              "marginTop": 13.5,
              "width": 600,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Light",
                "fontSize": 14,
                "letterSpacing": 0,
                "lineHeight": 22,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            British sovereign bonds are rallying strongly after the EU warned over the weekend that Boris Johnson faces an uphill battle in striking a Brexit.
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#E6ECED",
            "borderBottomWidth": 1,
            "width": 750,
          }
        }
      />
      <View
        style={
          Object {
            "paddingBottom": 75,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 26.5,
              "marginTop": 66.5,
              "width": 649.5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#262676",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 18,
                "letterSpacing": -0.12,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Sources
          </Text>
          <ForwardRef
            onPress={[Function]}
          >
            <Text
              style={
                Object {
                  "color": "#7B46E4",
                  "fontFamily": "BananaGrotesk-Bold",
                  "fontSize": 16,
                  "letterSpacing": -0.09,
                  "opacity": 1,
                  "textAlign": "right",
                }
              }
            >
              View all
            </Text>
          </ForwardRef>
        </View>
        <View
          style={
            Object {
              "marginVertical": 15,
            }
          }
        >
          <storyLineShortestCard
            date="23 hours ago"
            description="AS Boris Delivers A Brexit Deal, EU Have Agreed…"
            storyImage={
              Object {
                "default": "",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "marginVertical": 15,
            }
          }
        >
          <storyLineShortestCard
            date="23 hours ago"
            description="AS Boris Delivers A Brexit Deal, EU Have Agreed…"
            storyImage={
              Object {
                "default": "",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "marginVertical": 15,
            }
          }
        >
          <storyLineShortestCard
            date="23 hours ago"
            description="AS Boris Delivers A Brexit Deal, EU Have Agreed…"
            storyImage={
              Object {
                "default": "",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5F8F8",
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomColor": "#E6ECED",
              "borderBottomWidth": 1,
              "width": 750,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F8F8",
              "width": 750,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
                "width": 650,
                "zIndex": 100,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  Object {
                    "default": "",
                  }
                }
                style={
                  Object {
                    "marginRight": 10,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#8F8FAF",
                    "fontFamily": "BananaGrotesk-Regular",
                    "fontSize": 28,
                    "letterSpacing": -0.08,
                    "opacity": 1,
                    "textAlign": "left",
                  }
                }
              >
                154K Comments
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1F1F60",
                    "fontFamily": "BananaGrotesk-Regular",
                    "fontSize": 28,
                    "letterSpacing": -0.08,
                    "marginRight": 5,
                    "opacity": 1,
                    "textAlign": "left",
                  }
                }
              >
                Most Popular
              </Text>
              <DropDown2
                content={
                  Array [
                    "Most popular",
                    "Most recent",
                  ]
                }
                getOption={[Function]}
                headerTitle="Sort by"
                type={2}
                zIndex={1}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "#E6ECED",
                "borderBottomWidth": 1,
                "width": 750,
              }
            }
          />
          <View
            style={
              Object {
                "marginTop": 28,
              }
            }
          >
            <View
              key="0"
              style={
                Object {
                  "flexDirection": "row",
                  "marginBottom": 40,
                  "width": 647,
                }
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 5,
                    "marginTop": 15,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "default": "",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 14,
                      "height": 28,
                      "width": 28,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "width": 585.5,
                  }
                }
              >
                <commentBox
                  boostCount={450}
                  boosts={450}
                  commentText="When Robert Mueller broke his silence in May, his main point was that his long-awaited report spoke for itself."
                  isReply={false}
                  name="Jammie Rice"
                  onClickBoost={[Function]}
                  onClickReply={[Function]}
                  replyCount={1}
                  selectionHandler={[Function]}
                  time="32 Days"
                />
                <View
                  key="0"
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 20,
                      "width": "100%",
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "default": "",
                      }
                    }
                    style={
                      Object {
                        "borderRadius": 14,
                        "height": 28,
                        "width": 28,
                      }
                    }
                  />
                  <View
                    style={
                      Object {
                        "marginLeft": 5,
                        "width": 516,
                      }
                    }
                  >
                    <commentBox
                      boostCount={450}
                      boosts={450}
                      commentText="After reports emerged suggesting the Obama administration would issue additional sanctions imminently."
                      isReply={true}
                      name="Lorraine Kim"
                      onClickBoost={[Function]}
                      onClickReply={[Function]}
                      profilePic={
                        Object {
                          "default": "",
                        }
                      }
                      replyCount={1}
                      selectionHandler={[Function]}
                      time="32 Days"
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              key="1"
              style={
                Object {
                  "flexDirection": "row",
                  "marginBottom": 40,
                  "width": 647,
                }
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 5,
                    "marginTop": 15,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "default": "",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 14,
                      "height": 28,
                      "width": 28,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "width": 585.5,
                  }
                }
              >
                <commentBox
                  boostCount={450}
                  boosts={450}
                  commentText="When Robert Mueller broke his silence in May, his main point was that his long-awaited report spoke for itself."
                  isReply={false}
                  name="Abraham P."
                  onClickBoost={[Function]}
                  onClickReply={[Function]}
                  replyCount={3}
                  selectionHandler={[Function]}
                  time="32 Days"
                />
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 13,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#1F1C62",
                        "fontFamily": "BananaGrotesk-Bold",
                        "fontSize": 13,
                        "letterSpacing": -0.07,
                        "opacity": 1,
                        "textAlign": "left",
                      }
                    }
                  >
                    View 
                    3
                     Replies
                  </Text>
                </View>
              </View>
            </View>
            <View
              key="2"
              style={
                Object {
                  "flexDirection": "row",
                  "marginBottom": 40,
                  "width": 647,
                }
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 5,
                    "marginTop": 15,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "default": "",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 14,
                      "height": 28,
                      "width": 28,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "width": 585.5,
                  }
                }
              >
                <commentBox
                  boostCount={450}
                  boosts={450}
                  commentText="When Robert Mueller broke his silence in May, his main point was that his long-awaited report spoke for itself."
                  isReply={false}
                  name="Abraham P."
                  onClickBoost={[Function]}
                  onClickReply={[Function]}
                  replyCount={0}
                  selectionHandler={[Function]}
                  time="32 Days"
                />
              </View>
            </View>
            <View
              key="3"
              style={
                Object {
                  "flexDirection": "row",
                  "marginBottom": 40,
                  "width": 647,
                }
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 5,
                    "marginTop": 15,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "default": "",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 14,
                      "height": 28,
                      "width": 28,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "width": 585.5,
                  }
                }
              >
                <commentBox
                  boostCount={450}
                  boosts={450}
                  commentText="When Robert Mueller broke his silence in May, his main point was that his long-awaited report spoke for itself."
                  isReply={false}
                  name="Jammie Rice"
                  onClickBoost={[Function]}
                  onClickReply={[Function]}
                  replyCount={0}
                  selectionHandler={[Function]}
                  time="32 Days"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "height": 180,
          "width": 750,
        }
      }
    >
      <Commentbar2
        captureText={[Function]}
        onPress={[Function]}
        placeholderText="Comment Something"
      />
    </View>
  </ScrollViewMock>
</View>
`;
